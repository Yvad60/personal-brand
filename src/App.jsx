import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Button, Typography } from '@mui/material';
import { increment, decrement } from './redux/reducers/counterSlice.js';

function App() {
	const count = useSelector((state) => state.counter.value);
	const dispatch = useDispatch();

	return (
		<div>
			<Typography variant="h2">Sample counter app</Typography>
			<Typography variant="h2">{count}</Typography>
			<Button
				variant="contained"
				sx={{ mr: 3 }}
				size="large"
				onClick={() => dispatch(increment())}
			>
				Increment
			</Button>
			<Button
				variant="contained"
				size="large"
				color="error"
				onClick={() => dispatch(decrement())}
			>
				Decrement
			</Button>
		</div>
	);
}

export default App;
