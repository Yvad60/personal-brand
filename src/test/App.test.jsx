import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import React from 'react';
import App from '../App.jsx';
import store from '../redux/store.js';

describe('App tests', () => {
	test('should render app without crashing', () => {
		const wrapper = render(
			<Provider store={store}>
				<App />
			</Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});

	describe('components tests', () => {
		beforeEach(() => {
			render(
				<Provider store={store}>
					<App />
				</Provider>
			);
		});
		it('should contain app heading', () => {
			const headingText = screen.getByText(/Sample counter app/i);
			expect(headingText).toBeInTheDocument();
		});
		it('should display count of 0 by default', () => {
			const count = screen.getByText(/0/i);
			expect(count).toBeInTheDocument();
		});
		it('should increment on increment button click', () => {
			const incrementBtn = screen.getByText(/increment/i);
			fireEvent.click(incrementBtn);
			const count = screen.getByText(/1/i);
			expect(count).toBeInTheDocument();
		});
		it('should decrement on decrement button click', () => {
			const decrementBtn = screen.getByText(/decrement/i);
			fireEvent.click(decrementBtn);
			const count = screen.getByText(/0/);
			expect(count).toBeInTheDocument();
		});
	});
});
