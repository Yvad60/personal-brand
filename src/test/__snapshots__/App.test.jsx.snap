// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render app without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          Sample counter app
        </h2>
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          0
        </h2>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jmm4hg-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Increment
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-1i749hr-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Decrement
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        Sample counter app
      </h2>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        0
      </h2>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jmm4hg-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Increment
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-1i749hr-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Decrement
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
